src/ngamsPlugIns/ngamsPlugIns/ngamsSdmMultipart.py

```diff
@@ -58,168 +62,151 @@ def specificTreatment(fo):
                 The finalFileName is a string containing the name of the final
                 file without extension. type is the mime-type from the header.
     """
-    import rfc822, cgi, re
-    _EXT = '.msg'
-
-    filename = fo.name
-
-#    uidTempl = re.compile("[xX][0-9,a-f]{3,}/[xX][0-9,a-f]{1,}$")
-# This matches the old UID of type    X0123456789abcdef/X01234567
-
-#    uidTempl = re.compile("^[uU][iI][dD]:/(/[xX][0-9,a-f,A-F]+){3}(#\w{1,}|/\w{0,}){0,}$")
-# This matches the new UID structure uid://X1/X2/X3#kdgflf
-
-    # update for the new assignment of archiveIds (backwards compatible)
-    uidTempl = re.compile("^[uU][iI][dD]://[aAbBcCzZxX][0-9,a-z,A-Z]+(/[xX][0-9,a-z,A-Z]+){2}(#\w{1,}|/\w{0,}){0,}$")
 
+    filename = file_object.name
     try:
-        message = rfc822.Message(fo)
-        type, tparams = cgi.parse_header(message["Content-Type"])
-    except Exception as e:
-        err = "Parsing of mime message failed: " + str(e)
-        errMsg = genLog("NGAMS_ER_DAPI_BAD_FILE", [os.path.basename(filename),
-                                                   _PLUGIN_ID, err])
-        raise Exception(errMsg)
+        message = rfc822.Message(file_object)
```

Comment:
This would be something like `message = email.parse.<oneofheparsers>.parse(fille_object)`

Reply:
Thank you for the suggestion. I have applied the changes. Hopefully it is python 3 compatible.
I also decided it makes more sense to pass the file path instead of a file object. Opening a file object in a function and then passing it to another function seems like an inefficient way of doing things. The file is only opened for the duration of the parsing.